src/parser/transform/constraint/transform_constraint.cpp

```diff
@@ -61,6 +61,9 @@ unique_ptr<Constraint> Transformer::TransformConstraint(duckdb_libpgquery::PGLis
 	D_ASSERT(constraint);
 	switch (constraint->contype) {
 	case duckdb_libpgquery::PG_CONSTR_NOTNULL:
+		if (column.type.IsSerial()) {
```

Comment:
This might be an unnecessary check; I'm not sure if it's a bad thing for two identical constraints to be associated with a table (e.g. if someone declared like `pk_id SERIAL NOT NULL` etc.)

Reply:
I would probably copy Postgres behavior here